docs/security/tokens/using-tokens.md

```diff
@@ -0,0 +1,237 @@
+
+Use this document to learn how to request tokens from an OpenID Connect (OIDC) Provider or how to generate a test token
+for validating your OSG services.
+
+Requesting Tokens From An OIDC Provider
+---------------------------------------
+
+If you are a member of a collaboration with an OIDC provider (such as [CILogon](https://www.cilogon.org/oidc) or
+[Indigo IAM](https://indigo-iam.github.io/v/v1.7.2/docs/)), you can use the
+[oidc-agent](https://indigo-dc.gitbook.io/oidc-agent/) client to request tokens.
+This client tool is available either as [a container](#using-a-container) or as
+[an RPM installation](#using an RPM installation).
+
+!!! tip "Alternative tokens for testing"
+    If you are not a member of a collaboration with access to an OIDC provider, you can generate test SciTokens using
+    [these instructions](#generating-scitokens-for-testing)
+
+### Using a Container
+
+#### Registering an OIDC profile
+
+1. Start an agent container in the background and name it `my-agent` to easily run subsequent commands against it:
+
+        :::console
+        docker run -d --name my-agent opensciencegrid/oidc-agent:3.6-release
+
+1. Generate a local client profile and follow the prompts:
+
+        :::console
+        docker exec -it my-agent oidc-gen -w device <CLIENT PROFILE>
+
+    1. Specify an OIDC provider such as CILogon or an IAM instance as the client issuer.
+       For example, if you are requesting tokens from the WLCG IAM instance:
+
+            Issuer [https://iam-test.indigo-datacloud.eu/]: https://wlcg.cloud.cnaf.infn.it/
+
+    1. Request scopes for the capabilities that you need based on the type of tokens that your provider issues:
+
+        | **Capability**   | **SciTokens Scope** | **WLCG Scope**                                 |
+        |:-----------------|---------------------|------------------------------------------------|
+        | HTCondor `READ`  | `condor:/READ`      | `compute.read`                                 |
+        | HTCondor `WRITE` | `condor:WRITE`      | `compute.modify compute.cancel compute.create` |
+        | XRootD read      | `read:<PATH>`       | `storage.read:<PATH>`                          |
+        | XRootD write     | `write:<PATH>`      | `storage.modify:<PATH>`                        |
+
+        Replacing `<PATH>` with a path to the storage location that the bearer should be authorized to access.
+        If you are requesting WLCG tokens, you will need to also add the `wlcg` and `offline_access` scopes.
+        For example, to request HTCondor `READ` and `WRITE` access from an OIDC provider issuing WLCG tokens,
+        specify the following scopes:
+
+            This issuer supports the following scopes: openid profile email address phone offline_access wlcg iam wlcg.groups
+            Space delimited list of scopes or 'max' [openid profile offline_access]: wlcg offline_access compute.read compute.modify compute.cancel compute.create
```

Comment:
Is there a way to distinguish the prompt from the bit you have to type?  If not, instead of pasting the whole interaction you may just want to say

> When prompted by oidc-agent for a space delimited list of scopes, specify `wlcg offline_access compute.read compute.modify compute.cancel compute.create`

Reply:
Done